shelley/chain-and-ledger/executable-spec/test/Generator/Delegation/QuickCheck.hs

```diff
@@ -98,80 +120,118 @@ genDCerts keys coreKeys vrfKeys pparams dpState slot ttl = do
 -- and we generate more delegations than registrations of keys/pools.
 genDCert
   :: KeyPairs
+  -> MultiSigPairs
   -> [CoreKeyPair]
   -> VrfKeyPairs
   -> PParams
   -> DPState
   -> SlotNo
-  -> Gen (Maybe (DCert, Either KeyPair [CoreKeyPair]))
-genDCert keys coreKeys vrfKeys pparams dpState slot =
-  QC.frequency [ (2, genRegKeyCert keys dState)
+  -> Gen (Maybe (DCert, CertCred))
+genDCert keys scripts coreKeys vrfKeys pparams dpState slot =
+  QC.frequency [ (2, genRegKeyCert keys scripts dState)
                , (2, genRegPool keys vrfKeys dpState)
-               , (3, genDelegation keys dpState)
+               , (3, genDelegation keys scripts dpState)
                , (1, genGenesisDelegation keys coreKeys dpState)
-               , (1, genDeRegKeyCert keys dState)
+               , (1, genDeRegKeyCert keys scripts dState)
                , (1, genRetirePool keys pparams pState slot)
                , (1, genInstantaneousRewards coreKeys pparams dState)
                ]
  where
   dState = dpState ^. dstate
   pState = dpState ^. pstate
 
--- | Generate a RegKey certificate along and also returns the stake key
+-- | Generate a RegKey certificate along and also returns the staking credential
 -- (needed to witness the certificate)
 genRegKeyCert
   :: KeyPairs
+  -> MultiSigPairs
   -> DState
-  -> Gen (Maybe (DCert, Either KeyPair [CoreKeyPair]))
-genRegKeyCert keys delegSt =
-  case availableKeys of
-    [] -> pure Nothing
-    _ -> do
-           (_payKey, stakeKey) <- QC.elements availableKeys
-           pure $ Just (DCertDeleg (RegKey (toCred stakeKey)), Left stakeKey)
+  -> Gen (Maybe (DCert, CertCred))
+genRegKeyCert keys scripts delegSt =
+  QC.oneof
```

Comment:
is a 50/50 split between Scripts/Keys ok?

Reply:
tbh. no idea, but we can add those choices to the discussion / issue #1109